src/tree_cmd.cpp

```diff
@@ -178,6 +182,118 @@ static void PlaceTree(TileIndex tile, uint32_t r)
 	}
 }
 
+struct BlobHarmonic
+{
+	int amplitude;
+	float phase;
+	int frequency;
+};
+
+/**
+ * Creates a star-shaped[sic] polygon originating from (0, 0) as defined by the given harmonics.
+ *
+ * @param radius The maximum radius of the polygon. May be smaller, but will not be larger.
+ * @param harmonics a std::vector of the harmonics data.
+ * @param noOfSegments How many segments make up the polygon.
+ */
+std::vector<Point> CreateStarShapedPolygon(const int radius, const std::vector<BlobHarmonic> harmonics, const int noOfSegments)
+{
+	std::vector<Point> result;
```

Comment:
This one is probably fine as a vector, in which case you should add `result.reserve(noOfSegments);` so that only one allocation is made.

Reply:
i've already converted this to an array, but i can revert that if you wish.